src/main/java/com/dnd/runus/auth/oidc/provider/impl/AppleAuthProvider.java

```diff
@@ -112,19 +116,23 @@ private Claims parseClaims(String token, PublicKey publicKey) {
         }
     }
 
-    private void validateClaims(Claims claims) {
-        // todo 4가지 검증 추가
-        // 1. Verify the nonce for the authentication
-        // 2. Verify that the iss field contains https://appleid.apple.com
-        // 3. Verify that the aud field is the developer’s client_id
-        // 4. Verify that the time is earlier than the exp value of the token
-        // nonce 검증 방법 알아보기
-        //        if(claims.get("nonce") == null) {
-        //            throw new AuthException(ErrorType.TAMPERED_ACCESS_TOKEN,
-        //        }
-        //        if (!claims.getIssuer().contains("https://appleid.apple.com")) {
-        //            throw new AuthException(ErrorType.TAMPERED_ACCESS_TOKEN, "잘못된 iss");
-        //        }
+    private void verifyClaims(Claims claims) {
+
+        if (claims.get("nonce") == null) {
+            throw new AuthException(ErrorType.TAMPERED_ACCESS_TOKEN, "Can't verify the nonce");
+        }
+        if (!claims.getIssuer().contains("https://appleid.apple.com")) {
+            throw new AuthException(ErrorType.TAMPERED_ACCESS_TOKEN, "Can't verify iss");
+        }
+        if (!claims.getAudience().contains(clientId)) {
+            throw new AuthException(ErrorType.TAMPERED_ACCESS_TOKEN, "Can't verify audience");
+        }
+
+        Instant exp = Instant.ofEpochMilli(claims.getExpiration().getTime());
+        Instant now = Instant.now();
+        if (now.isAfter(exp)) {
```

Comment:
```suggestion
        if (Instant.now().isAfter(exp)) {
```

한줄로 적으면 더 간단해질 것 같아요!